ros/api/common/add_group_filter.py

```diff
@@ -7,16 +7,33 @@
 
 
 def group_filtered_query(query):
+    all_host_groups = get_host_groups()
+    host_groups_without_none = [group for group in all_host_groups if group is not None]
     if get_flag_value(FLAG_INVENTORY_GROUPS):
-        if len(get_host_groups()) >= 1:
-            query = query.filter(System.groups[0]['id'].astext.in_(get_host_groups()) | (System.groups == '[]'))
+        if len(all_host_groups) >= 1:
+            if load_ungrouped_sys(all_host_groups):
+                query = query.filter(System.groups[0]['id'].astext.in_(host_groups_without_none)
+                                     | (System.groups == '[]'))
+            elif not load_ungrouped_sys(all_host_groups):
+                query = query.filter(System.groups[0]['id'].astext.in_(host_groups_without_none))
+        elif len(all_host_groups) == 1 and load_ungrouped_sys(all_host_groups):
```

Comment:
In `if` loop, you are checking for `len(all_host_groups) >= 1`, when it will execute `elif`block?